test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Extensions/Binders/ApplyBinderTests.cs

```diff
@@ -757,6 +757,58 @@ public void BindVirtualPropertiesAfterCollapseReturnsApplyClause()
             Assert.Equal(AggregationMethod.VirtualPropertyCount, aggExp.Method);
         }
 
+        [Fact]
+        public void BindApplyWithGroupByAndTypeCastShouldReturnApplyClause()

```

Comment:
These two tests seem very similar 
Could they not be made one parameterized test ?

Reply:
The assertions are not the same. I don't really see how to merge these tests into a parameterizable test without having to use conditions (which makes the code less understandable).